### PR TITLE
topology: switch some SDW topologies to use dynamic pipelines

### PR DESCRIPTION
**Pull request by ranj063**
We've been testing the HDA and some nocodec topologies for a few weeks now. Switch some more now to increase testing coverage.

### CODE REVIEW

File: tools/topology/CMakeLists.txt
Comment (plbossart) [inline tools/topology/CMakeLists.txt:89]: let's also convert from line 85

File: tools/topology/CMakeLists.txt
Comment (plbossart) [inline tools/topology/CMakeLists.txt:91]: let's also convert until line 103 included.

File: tools/topology/CMakeLists.txt
Comment (plbossart) [inline tools/topology/CMakeLists.txt:118]: same, let's add lines 119 and 120

**Review by plbossart — CHANGES_REQUESTED** [review]
I'd prefer all SoundWire topologies to be converted in one shot. it'll be easier to validate and find bugs. Everything that has references to ALH should be converted.

File: tools/topology/CMakeLists.txt
Comment (ranj063) [reply tools/topology/CMakeLists.txt:118]: All added @plbossart

**Comment by plbossart** [comment]
since most SoundWire devices are not in the CI farm, we need to ask @fredoh9 and @aiChaoSONG for a dedicated SoundWire 'daily test' with this PR before merging it. Thanks!

**Comment by plbossart** [comment]
> Merge pending test results.

At this point the results on SoundWire platforms *before* this change are rather problematic so I'd really want to punt until the known issues are fixed.

**Comment by marc-hb** [comment]
> We've been testing the HDA and some nocodec topologies for a few weeks now.

See regressions in #4199 and #4273

**Comment by lgirdwood** [comment]
@ranj063 I'm assuming that https://github.com/thesofproject/sof/commit/fcaede7a8710c94a774aa080c425774d517ce43e will fix this ?

**Comment by ranj063** [comment]
> @ranj063 I'm assuming that [fcaede7](https://github.com/thesofproject/sof/commit/fcaede7a8710c94a774aa080c425774d517ce43e) will fix this ?

yes that will fix #4199 and I have a kernel fix that will fix #4273. But Im not sure if we'd still have to wait for the other SDW bugs before merging this

**Comment by plbossart** [comment]
> > @ranj063 I'm assuming that [fcaede7](https://github.com/thesofproject/sof/commit/fcaede7a8710c94a774aa080c425774d517ce43e) will fix this ?
> 
> yes that will fix #4199 and I have a kernel fix that will fix #4273. But Im not sure if we'd still have to wait for the other SDW bugs before merging this

yes, no SoundWire changes until we are back to normal in the daily tests, thanks!

**Comment by lgirdwood** [comment]
> > @ranj063 I'm assuming that [fcaede7](https://github.com/thesofproject/sof/commit/fcaede7a8710c94a774aa080c425774d517ce43e) will fix this ?
> 
> yes that will fix #4199 and I have a kernel fix that will fix #4273. But Im not sure if we'd still have to wait for the other SDW bugs before merging this

Are these known SDW bugs or yet to be discovered ? i.e. would merging this now expose the bugs since we have time before v1.9. 

@plbossart are you happy to merge or do you have pending SDW issues to fix first for SDW stability?

**Comment by ranj063** [comment]
> Are these known SDW bugs or yet to be discovered ? i.e. would merging this now expose the bugs since we have time before v1.9.

these are known bugs today. But the reason I want to enable dynamic pipelines on these is to expose new bugs that it might bring before 1.9

**Comment by plbossart** [comment]
> > Are these known SDW bugs or yet to be discovered ? i.e. would merging this now expose the bugs since we have time before v1.9.
> 
> these are known bugs today. But the reason I want to enable dynamic pipelines on these is to expose new bugs that it might bring before 1.9

@ranj063 @fredoh9 we need to have a dedicated test run for this PR, i asked for this 8 days ago :-)

**Comment by lgirdwood** [comment]
> @ranj063 @fredoh9 we need to have a dedicated test run for this PR, i asked for this 8 days ago :-)

Ah, great. :) @fredoh9 are you doing this or maybe @XiaoyunWu6666  ?

**Comment by fredoh9** [comment]
internal daily test 4364 is triggered with kernel topic/sof-dev tip
`20cf8377 fixup! ASoC: SOF: Don't set up widgets during topology parsing`

**Comment by lgirdwood** [comment]
@fredoh9 @ranj063 are we good to merge or blocking on tests passing ?

**Comment by ranj063** [comment]
> @fredoh9 @ranj063 are we good to merge or blocking on tests passing ?

@lgirdwood not ready yet. SDW issues are still being worked upon. Lets hold off merging this for another 2-3 days until daily tests stabilize

**Comment by plbossart** [comment]
@ranj063 this one should be updated, we should really have all the HDAudio, SoundWire and nocodec topologies used with DYNAMIC pipelines.

**Comment by lgirdwood** [comment]
> @ranj063 this one should be updated, we should really have all the HDAudio, SoundWire and nocodec topologies used with DYNAMIC pipelines.

Ack - a good v2.1 feature :)

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

**Comment by lgirdwood** [comment]
@plbossart @ranj063 any blockers for this now ?

**Comment by plbossart** [comment]
@lgirdwood @ranj063 @marc-hb I don't think we should change topologies, it could very well be that they end-up being used with a kernel that's too old. We've already seen this problem with HDaudio.

I think we need to enable dynamic pipelines in our CI tests though, so maybe this needs to be combined with the deep-buffer changes I suggested, that way we test the latest and greatest topologies but we don't impact existing end-user setups.

Alternatively, we can use a kernel parameter to force all topologies used by CI to be dynamic, which means there is no need to change files one by one.

**Comment by lgirdwood** [comment]
> @lgirdwood @ranj063 @marc-hb I don't think we should change topologies, it could very well be that they end-up being used with a kernel that's too old. We've already seen this problem with HDaudio.
> 
> I think we need to enable dynamic pipelines in our CI tests though, so maybe this needs to be combined with the deep-buffer changes I suggested, that way we test the latest and greatest topologies but we don't impact existing end-user setups.
> 
> Alternatively, we can use a kernel parameter to force all topologies used by CI to be dynamic, which means there is no need to change files one by one.

Ok, will close this so that SDW can be folded into the CI at first and changes can be tested incrementally..

**Pull request closed without merging**
